src/morphological_operations.jl

```diff
@@ -99,3 +99,90 @@ which is defined as the arithmetic difference between the internal and the exter
 `region` allows you to control the dimensions over which this operation is performed.
 """
 morpholaplace(img::AbstractArray, region=coords_spatial(img)) = dilate(img, region) + erode(img, region) - 2img
+
+
+struct FalsePadded
+    img::AbstractArray{Bool,2}
+end
+import Base.getindex
+function getindex(img::FalsePadded, r::Integer, c::Integer)
+    if (r==0) || (c==0)
+        return false
+    elseif (r> size(img.img,1)) || (c>size(img.img,2))
+        return false
+   else
+        return img.img[r,c]
+   end
+end
+
+"""
+```
+function thinning_iteration(img::AbstractArray{Bool,2}; odd_iteration::Bool)
+```
+The thining iteration evaluates three conditions in order to determine which pixels of the image should be removed from img.
+The three conditions are explained in:
+Guo, Z., & Hall, R. W. (1989). Parallel thinning with two-subiteration algorithms. Communications of the ACM, 32(3), 359-373.
+"""
+function thinning_iteration!(img_ori::AbstractArray{Bool,2}, odd_iteration::Bool)
+    local marker = trues(size(img_ori))
```

Comment:
I think we can remove all occurrences of the keyword `local` in the implementation. Did you choose to use `local` in these places for a reason?

Reply:
Yes. They can be removed.  maker is local no matter what, because of the assignment. I think I wrote it as an habit I should get rid off.

Reply:
Yes, `local` has a different purpose in the language, to preserve the value of an outside variable in a loop of the same name.